File: Elixir/problem020.exs
# Problem 20 - Project Euler
# http://projecteuler.net/index.php?section=problems&id=20
defmodule P020 do
  def solve() do
    1..100
    |> Enum.reduce(1, &(&1 * &2))
    |> Integer.digits
    |> Enum.sum
  end
end

IO.puts P020.solve()

File: Kotlin/problem020.kt
// Problem 20 - Project Euler
// http://projecteuler.net/index.php?section=problems&id=20
fun sum_of_digits_factorial(x: Int): Int {
    return (1..x).fold(1.toBigInteger(), { p, i -> p * i.toBigInteger() })
            .toString()
            .map { it.code - 48 }
            .sum()
}

fun main() {
    println(sum_of_digits_factorial(100))
}
